maa-cli/src/run/copilot.rs

```diff
@@ -0,0 +1,207 @@
+use std::{
+    fs::{self, File},
+    io::{self, BufReader, Write},
+};
+
+use super::{run, Result};
+use crate::{
+    config::task::{
+        default_variants, task_type::TaskType, value::input::Input, Strategy, Task, TaskList, Value,
+    },
+    dirs::Dirs,
+    installer::maa_core::find_resource,
+    object,
+};
+use anyhow::Context;
+use reqwest::blocking::Client;
+use serde_json::Value as JsonValue;
+
+pub fn copilot(dirs: &Dirs, link: String) -> bool {
+    let api = "https://prts.maa.plus/copilot/get/";
+    let cache_dir = dirs.cache().display().to_string();
+    let base_resource_dir = find_resource(dirs).context("Failed to find resource!");
+    let stage_dir = format!(
```

Comment:
For joining paths in Rust, it's better to use `path1.join(path2)` instead of converting it to a string and then formatting.